addons/mass_mailing/models/mailing.py

```diff
@@ -1127,12 +1127,19 @@ def convert_links(self):
     @api.model
     def _process_mass_mailing_queue(self):
         mass_mailings = self.search([('state', 'in', ('in_queue', 'sending')), '|', ('schedule_date', '<', fields.Datetime.now()), ('schedule_date', '=', False)])
+        remaining_mass_mailings = len(mass_mailings)
+        self.env['ir.cron']._notify_progress(0, remaining_mass_mailings)
         for mass_mailing in mass_mailings:
+            remaining_mass_mailings -= 1
+            self.env['ir.cron']._notify_progress(1)
             context_user = mass_mailing.user_id or mass_mailing.write_uid or self.env.user
             mass_mailing = mass_mailing.with_context(
                 **self.env['res.users'].with_user(context_user).context_get()
             )
-            if len(mass_mailing._get_remaining_recipients()) > 0:
+
+            remaining_mailing_recipient_count = len(mass_mailing._get_remaining_recipients())
+            if remaining_mailing_recipient_count > 0:
+                self.env['ir.cron']._notify_progress(0, remaining_mass_mailings + remaining_mailing_recipient_count)
```

Comment:
I find new code complicated to read and understand and I don't see what we are gaining ... what is the purpose of managing mailings with the progress ? Mailings already have a way to know if they are in sending mode or not; and anyway the cron should run frequently while not bieng critical. What are we gaining here ?

Reply:
Same as for the mail cron, this is temporary to ensure there is no breaking change to cron who use the "retrigger on timeout" "feature", I'll go through each team CRON with ppl from those team to fully apply the new guidelines